R/graded.R

```diff
@@ -644,10 +644,12 @@ fail_if_code_feedback <- function(
   if (is_placeholder(user_code, ".user_code")) {
     user_code <- get_from_env(".user_code", env)
     assert_object_found_in_env(user_code, env, "fail_if_code_feedback")
-    if (is.null(user_code) || length(user_code) == 0 || !nzchar(user_code)) {
-      graded(logical(), "I didn't receive your code. Did you write any?", type = "info")
-    }
   }
+
+  if (length(user_code) == 0 || is_empty_code(user_code)) {
+    return(grade_code_is_empty())
+  }
```

Comment:
Did you mean to check if `user_code` is NULL here?

```suggestion
  if (is.null(user_code) || is_empty_code(user_code)) {
    return(grade_code_is_empty())
  }
```

Reply:
Actually `is_empty_code(NULL)` should also return `TRUE`, so that's probably all you need here.